FollowStatsIT#testFollowStatsApiIncludeShardFollowStatsWithRemovedFollowerIndex failure

```
[2019-02-11T23:27:48,513][INFO ][o.e.c.m.MetaDataDeleteIndexService] [node_s_0] [leader1/CauqUnDjRpivCn8-gN51hQ] deleting index
FAILURE 0.49s J6 | FollowStatsIT.testFollowStatsApiIncludeShardFollowStatsWithRemovedFollowerIndex <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: 
   > Expected: <1>
   >      but: was <0>
   >    at __randomizedtesting.SeedInfo.seed([A9F2E2FB2B3232F2:982741CACE61265E]:0)
   >    at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   >    at org.elasticsearch.xpack.ccr.FollowStatsIT.testFollowStatsApiIncludeShardFollowStatsWithRemovedFollowerIndex(FollowStatsIT.java:182)
   >    at java.lang.Thread.run(Thread.java:748)
```

Build url: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+g1gc/355/console

--- a/x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowNodeTask.java
+++ b/x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowNodeTask.java
@@ -452,9 +452,13 @@ public abstract class ShardFollowNodeTask extends AllocatedPersistentTask {
                 scheduler.accept(TimeValue.timeValueMillis(delay), task);
             }
         } else {
-            fatalException = ExceptionsHelper.convertToElastic(e);
-            LOGGER.warn("shard follow task encounter non-retryable error", e);
+            setFatalException(e);
         }
+    }
+
+    void setFatalException(Exception e) {
+        fatalException = ExceptionsHelper.convertToElastic(e);
+        LOGGER.warn("shard follow task encounter non-retryable error", e);
     }
 
     static long computeDelay(int currentRetry, long maxRetryDelayInMillis) {

--- a/x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowTasksExecutor.java
+++ b/x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowTasksExecutor.java
@@ -282,7 +282,7 @@ public class ShardFollowTasksExecutor extends PersistentTasksExecutor<ShardFollo
                     shardFollowNodeTask), e);
                 threadPool.schedule(() -> nodeOperation(task, params, state), params.getMaxRetryDelay(), Ccr.CCR_THREAD_POOL_NAME);
             } else {
-                shardFollowNodeTask.markAsFailed(e);
+                shardFollowNodeTask.setFatalException(e);
             }
         };
 

--- a/x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/FollowStatsIT.java
+++ b/x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/FollowStatsIT.java
@@ -149,7 +149,6 @@ public class FollowStatsIT extends CcrSingleNodeTestCase {
         assertAcked(client().execute(PauseFollowAction.INSTANCE, new PauseFollowAction.Request("follower1")).actionGet());
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/38779")
     public void testFollowStatsApiIncludeShardFollowStatsWithRemovedFollowerIndex() throws Exception {
         final String leaderIndexSettings = getIndexSettings(1, 0,
             singletonMap(IndexSettings.INDEX_SOFT_DELETES_SETTING.getKey(), "true"));

